controller/handler/badge.go

```diff
@@ -209,17 +216,17 @@ func generateResponseTimeBadgeSVG(duration string, averageResponseTime int) []by
 	return svg
 }
 
-func getBadgeColorFromResponseTime(responseTime int) string {
-	if responseTime <= 50 {
-		return badgeColorHexAwesome
-	} else if responseTime <= 200 {
-		return badgeColorHexGreat
-	} else if responseTime <= 300 {
-		return badgeColorHexGood
-	} else if responseTime <= 500 {
-		return badgeColorHexPassable
-	} else if responseTime <= 750 {
-		return badgeColorHexBad
+func getBadgeColorFromResponseTime(responseTime int, key string, config *config.Config) string {
+
+	group, name, _ := store.Get().GetGroupAndNameByKey(key)
+	ep := config.GetEndpointByGroupAndName(group, name)
+
+	// the threshold config requires 5 values, so we can be sure it's set here
+	for i := 0; i < 5; i++ {
+		if responseTime <= ep.UIConfig.Badge.ResponseTime.Thresholds[i] {
+			return badgeColors[i]
+		}
 	}
```

Comment:
I don't know how I feel about the fact that to retrieve the badge configuration, we need to make a call to the database.
Instead, `config` should have a function `GetEndpointByKey`. 

The key is generated using `ConvertGroupAndEndpointNameToKey(group, name)`, so it should be possible to just iterate through the endpoints and compare the given key with the each endpoints' key.